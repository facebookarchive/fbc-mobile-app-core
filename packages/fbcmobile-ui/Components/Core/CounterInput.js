/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import * as React from 'react';
import Text from '@fbcmobile/ui/Components/Core/Text';
import {Colors} from '@fbcmobile/ui/Theme';
import {Icon} from 'react-native-material-ui';
import {StyleSheet, TouchableOpacity, View} from 'react-native';

type Props = {
  +count: number,
  +minCount: number,
  +maxCount: number,
  +onCountChanged: (newCount: number) => void,
};

const CounterInput = ({count, minCount, maxCount, onCountChanged}: Props) => {
  return (
    <View style={styles.root}>
      <TouchableOpacity
        disabled={count <= minCount}
        onPress={() => onCountChanged(count - 1)}>
        <Icon
          name="minus-circle-outline"
          iconSet="MaterialCommunityIcons"
          size={30}
          color={count <= minCount ? Colors.Gray85 : Colors.BlueGray}
        />
      </TouchableOpacity>
      <Text style={styles.count} variant="h2" weight="regular" color="gray">
        {count < 10 ? `0${count}` : count}
      </Text>
      <TouchableOpacity
        disabled={count >= maxCount}
        onPress={() => onCountChanged(count + 1)}>
        <Icon
          name="plus-circle-outline"
          iconSet="MaterialCommunityIcons"
          size={30}
          color={count >= maxCount ? Colors.Gray85 : Colors.BlueGray}
        />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    display: 'flex',
    alignItems: 'center',
    flexDirection: 'row',
  },
  count: {
    marginHorizontal: 10,
  },
});

export default CounterInput;
