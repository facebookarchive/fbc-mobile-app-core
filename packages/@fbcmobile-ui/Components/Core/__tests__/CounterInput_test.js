/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import CounterInput from '../CounterInput';
import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

describe('CounterInput Tests', () => {
  test('CounterInput renders correctly', () => {
    const onCountChanged = jest.fn();
    const tree = renderer
      .create(
        <CounterInput
          count={1}
          minCount={0}
          maxCount={10}
          onCountChanged={onCountChanged}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('CounterInput renders correctly with min count', () => {
    const onCountChanged = jest.fn();
    const tree = renderer
      .create(
        <CounterInput
          count={0}
          minCount={0}
          maxCount={10}
          onCountChanged={onCountChanged}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('CounterInput renders correctly with max count', () => {
    const onCountChanged = jest.fn();
    const tree = renderer
      .create(
        <CounterInput
          count={10}
          minCount={0}
          maxCount={10}
          onCountChanged={onCountChanged}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('CounterInput test onPress minus icon', () => {
    const onCountChanged = jest.fn();
    const {getByTestId} = render(
      <CounterInput
        count={10}
        minCount={0}
        maxCount={10}
        onCountChanged={onCountChanged}
      />,
    );
    const minusIcon = getByTestId('minus-icon');
    fireEvent.press(minusIcon);
    expect(onCountChanged).toHaveBeenCalled();
  });
});
