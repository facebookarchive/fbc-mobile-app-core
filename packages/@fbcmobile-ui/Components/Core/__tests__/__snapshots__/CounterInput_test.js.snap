// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterInput Tests CounterInput renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="minus-icon"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 30,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰍷
    </Text>
  </View>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 20,
            "fontWeight": "500",
            "lineHeight": 25,
          },
          Object {
            "color": "#303846",
          },
          Object {
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "marginHorizontal": 10,
        },
      ]
    }
  >
    01
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="plus-icon"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 30,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰐙
    </Text>
  </View>
</View>
`;

exports[`CounterInput Tests CounterInput renders correctly with max count 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="minus-icon"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 30,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰍷
    </Text>
  </View>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 20,
            "fontWeight": "500",
            "lineHeight": 25,
          },
          Object {
            "color": "#303846",
          },
          Object {
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "marginHorizontal": 10,
        },
      ]
    }
  >
    10
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="plus-icon"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#D8D8D8",
            "fontSize": 30,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰐙
    </Text>
  </View>
</View>
`;

exports[`CounterInput Tests CounterInput renders correctly with min count 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="minus-icon"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#D8D8D8",
            "fontSize": 30,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰍷
    </Text>
  </View>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 20,
            "fontWeight": "500",
            "lineHeight": 25,
          },
          Object {
            "color": "#303846",
          },
          Object {
            "fontFamily": "Roboto-Regular",
          },
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "marginHorizontal": 10,
        },
      ]
    }
  >
    00
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="plus-icon"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 30,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰐙
    </Text>
  </View>
</View>
`;
